Sources/Basics/Environment/Environment.swift

```diff
@@ -0,0 +1,326 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+
+#if USE_IMPL_ONLY_IMPORTS
+@_implementationOnly import TSCclibc
+#else
+import TSCclibc
+#endif
+
+// FIXME: Use Synchronization.Mutex when available
+class Mutex<T>: @unchecked Sendable {
+    var lock: NSLock
+    var value: T
+
+    init(value: T) {
+        self.lock = .init()
+        self.value = value
+    }
+
+    func withLock<U>(_ body: (inout T) -> U) -> U {
+        self.lock.lock()
+        defer { self.lock.unlock() }
+        return body(&self.value)
+    }
+}
+
+// FIXME: This should come from Foundation
```

Comment:
I don't think that this can come from Foundation. Foundation's API is not easy to change and the environment being case insensitive is not currently well handled there.

Reply:
The intent of the comment that this type _should_ come from a common Swift ecosystem library and foundation seems like the most natural place.